encoding/wkt/wkt.go

```diff
@@ -57,5 +59,10 @@ func Marshal(g geom.T, applyOptFns ...EncodeOption) (string, error) {
 
 // Unmarshal translates a WKT to the corresponding geometry.
 func Unmarshal(wkt string) (geom.T, error) {
```

Comment:
i think in go-geom, we want to relax things we check in CockroachDB (subject to @twpayne guidance!):
a) geometrycollections can have mixed dimensions.
b) linestrings/polygons do not have have 2/4 points respectively, and polygon rings don't have to be closed.

we can choose to optionally allow these checks (but have them off by default). this should be done with options -- see how we did the `EncodeOption` for `Marshal` above.

Reply:
> a) geometrycollections can have mixed dimensions.

Does this only apply to `GEOMETRYCOLLECTION` or does it apply to `GEOMETRYCOLLECTION{M, Z, ZM}` as well?

Reply:
Very good points! I've opened #200 to continue the discussion.